internal/clusterfeature/features/monitoring/operator.go

```diff
@@ -154,7 +207,7 @@ func (op FeatureOperator) Deactivate(ctx context.Context, clusterID uint, spec c
 		}
 	}
 
-	if boundSpec.Prometheus.SecretId == "" {
+	if boundSpec.Prometheus.Ingress.SecretId == "" {
```

Comment:
This isn't gonna work. Generated secret IDs should be persisted in the spec.

Reply:
this will be fixed in another PR

Reply:
this will be fixed in another PR